tests/lib/rules/jsx-embed-condition.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @fileoverview Prevents usage of && condition in JSX Embeds.
+ * @author Anees Iqbal <i@steelbrain.me>
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/jsx-embed-condition');
+
+const parserOptions = {
+  ecmaVersion: 2018,
+  sourceType: 'module',
+  ecmaFeatures: {
+    jsx: true
+  }
+};
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({parserOptions});
+ruleTester.run('jsx-embed-condition', rule, {
+  valid: [{
+    code: '<App>Test</App>'
+  }, {
+    code: '<App test>Another</App>'
+  }, {
+    code: '<App foo={e => bar(e)}>Hello World</App>'
+  }, {
+    code: '<App>{x ? <div></div> : null}</App>'
+  }, {
+    code: '<App>{x ? <div>Hello</div> : null}</App>'
+  }, {
+    code: '<App>{x ? <div>{y ? <y /> : <z />}</div> : null}</App>'
+  }, {
+    code: '<App x={x && y}>{x ? <div>{y ? <y /> : <z />}</div> : null}</App>'
+  }
+  // {
+  //   code: '<App test>{x ?? <div />}</App>'
+  // }
+  // TODO: ^ Using above gives a Parsing error in ESLint.
+  // Is the parser out of date
```

Comment:
no, but you can specify the `parserOptions.ecmaVersion` to be `2020` in this test case.

Reply:
to clarify, i meant in this *one test case*, not for all test cases.